plugin.py

```diff
@@ -0,0 +1,144 @@
+import sublime
+
+import os
+import re
+import base64
+from urllib.parse import urlparse
+import weakref
+
+from LSP.plugin.core.protocol import Request, Location
+from LSP.plugin.core.typing import Any, Callable, List, Mapping, Optional, Tuple, Union
+from LSP.plugin.core.registry import LspTextCommand
+from LSP.plugin import register_plugin, unregister_plugin
+from lsp_utils import NpmClientHandler
+
+class LspGrammarlyCommand(LspTextCommand):
+    session_name = 'grammarly'
+    msg_prefix = "LSP-Grammarly: "
+
+    def error_message(self, msg: str) -> None:
+        sublime.error_message(self.msg_prefix + msg)
+
+    def message_dialog(self, msg: str) -> None:
+        sublime.message_dialog(self.msg_prefix + msg)
+
+class LspGrammarlyExecuteLoginCommand(LspGrammarlyCommand):
+    cmd = "$/getOAuthUrl"
+    cmd_callback = "$/handleOAuthCallbackUri"
+
+    internal_redirect_uri = "vscode://znck.grammarly/auth/callback"
+    external_redirect_uri = internal_redirect_uri
+    redirect_uri = internal_redirect_uri
+
+    def run(self, edit: sublime.Edit) -> None:
+        def run_async() -> None:
+            session = self.session_by_name()
+            self.weaksession = weakref.ref(session)
```

Comment:
I will just push a couple of minor lint fixes and one fix here where `weakref.ref` could in theory be created with `None` value and crash.